exec/java-exec/src/main/java/org/apache/drill/exec/work/metadata/MetadataProvider.java

```diff
@@ -347,9 +351,9 @@ protected Response runInternal(final UserSession session, final SchemaTreeProvid
       } catch (Exception e) {
```

Comment:
We should change this to Throwable or keep the response sending code in finally block to avoid client waiting indefinitely.